docs/DafnyRef/UserGuide.md

```diff
@@ -156,14 +156,227 @@ exit code.
+
+[^caveat-about-assertion-and-assumption]: Caveat about assertion and assumption: One big difference between an "assertion transformed in an assumption" and the original "assertion" is that the original "assertion" can unroll functions twice, whereas the "assumed assertion" can unroll them only once. Hence, Dafny can still continue to analyze assertions after a failing assertion without automatically proving "false" (which would make all further assertions vacuous).
+
+[^smaller-batches]: To create a smaller batch, Dafny duplicates the assertion batch, and transforms each assertion into an assumption into exactly one batch, so that assertions are verified only in one batch. This results in "easier" formulas for the verifier because it has less to prove, but it takes more overhead because every verification instance have a common set of axioms and there is no knowledge sharing between instances because they run independently.
+
+### 24.8.3. Controlling assertion batches {#sec-assertion-batches-control}
+
+Here is how you can control how Dafny partition assertions into batches.
+
+* [`{:focus}`](#sec-focus) on an assert generates a separate assertion batch for the assertions of the enclosing block.
+* [`{:split_here}`](#sec-split_here) on an assert generates a separate assertion batch for assertions after this point.
+* [`{:vcs_split_on_every_assert}`](#sec-vcs_split_on_every_assert) on a function or a method generates one assertion batch per assertion
+
+You can also provide _heuristics_ for the verifier to split assertion batches, if you are not using the fine-grained attributes above. The effect of these attributes may vary, because they are low-level attributes and tune low-level heuristics.
+* [`{:vcs_max_cost N}`](#sec-vcs_max_cost) on a function or method enables splitting the assertion batch until the "cost" of each batch is below N.
+  Usually, you would set [`{:vcs_max_cost 0}`](#sec-vcs_max_cost) and [`{:vcs_max_splits N}`](#sec-vcs_max_splits) to ensure it generates N assertion batches.
+* [`{:vcs_max_keep_going_splits N}`](#sec-vcs_max_keep_going_splits) where N > 1 on a method dynamically splits the initial assertion batch up to N components if the verifier is stuck the first time.
+
+### 24.8.4. Command-line options and other attributes to control verification {#sec-command-line-options-and-attributes-for-verification}
```

Comment:
This section would be a great spot to mention the `dafny-reportgenerator` too, although that can wait for the next PR (since we'd want some background content on verification stability as well to motivate it)